mantidproject/mantid
# Remove unused Pylint configuration

### Description of work
Remove all Pylint config and tools

#### Purpose of work
We no longer use Pylint, and a lot of the functionality is now covered by ruff.

Fixes #37860 

#### Further detail of work
- I couldn't find any reference to Pylint in the docs/dev-docs, so nothing has been removed from docs.
- The leeroy.md file is still out of date but there is another issue to update that. Here I have just removed the Pylint stuff.
- The Pylint suppressions have been intentionally left in place because they may work with the ruff linter.

### To test:
Make sure the builds are fine.

*This does not require release notes* because **it's a removal of config for a tool we no longer use**



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
